src/support.js

```diff
@@ -246,14 +246,40 @@ Module['registerFunctions'] = registerFunctions;
 #endif // RELOCATABLE
 #endif // EMULATED_FUNCTION_POINTERS
 
+#if WASM_BACKEND
+#if RESERVED_FUNCTION_POINTERS
+var jsCallStartIndex = {{{ JSCALL_START_INDEX }}};
+var jsCallSigOrder = {{{ JSON.stringify(JSCALL_SIG_ORDER) }}};
+var jsCallNumSigs = Object.keys(jsCallSigOrder).length;
+var functionPointers = new Array(jsCallNumSigs * {{{ RESERVED_FUNCTION_POINTERS }}});
+#else
+var jsCallStartIndex = 1;
+var functionPointers = new Array({{{ RESERVED_FUNCTION_POINTERS }}});
```

Comment:
Isn't this branch `#if not RESERVED_FUNCTION_POINTERS`? What happens here in that case?

Reply:
In that case `RESERVED_FUNCTION_POINTERS` is 0, so `functionPointers` array will have the size of 0. That's the same behavior as the existing asm.js/fastcomp code.

Reply:
Oh ok, combining these into `WASM_BACKEND_WITH_RESERVED_FUNCTION_POINTERS` made this much clearer.

I was thinking "why not just say `new Array(0)` if we know it's 0", but that was because it was the
```
#if A
#if B
X
#else
Y
#endif
#else
Y
#endif
```
pattern, instead of the
```
#if A || B
X
#else
Y
#endif
```
pattern we wound up with.